env/fs_posix.cc

```diff
@@ -620,9 +620,10 @@ class PosixFileSystem : public FileSystem {
       }
     }
 
-    const auto pre_read_errno = errno;  // errno may be modified by readdir
+    // reset errno before calling readdir()
+    errno = 0;
```

Comment:
At this point, we know:
1) `d` is not null, otherwise we must have returned from this function already.
2) If `d` is not null here, should we also check the errno here, or can we assume it's 0?

Reply:
I think we cannot assume it's 0, as success lib function call could still change `errno`: https://stackoverflow.com/questions/45103482/can-errno-be-set-on-success
here we just reset it 0, as previous function call (`opendir()`) succeed, so we don't care what `errno` it set.